src/index.tsx

```diff
@@ -53,4 +55,12 @@ if (process.env.NODE_ENV !== 'production') {
   axe(React, ReactDOM, 1000, config);
 }
 
+if (process.env.PREVIEW === 'true') {
+  // Start Mirage
+  startMirage();
+}
+
+// Initialize services
+initilizeServices();
```

Comment:
If I'm understanding the reasoning behind this `init()` refactor, I think it could be avoided if we ensured that Mirage was always loaded before React was, and therefore before any of the Services could be instantiated. That's similar to what my patch for the live demo on the upstream website did.

I think it could be possible to achieve the same thing here by using some Webpack plugin to check for `process.env.NODE_ENV=='preview'` (or `process.env.PREVIEW=true`) and patch the `src/index.html` to add a `<script>` that loads Mirage. What do you think?

Reply:
Or perhaps instead of directly patching the `index.html`, using the Webpack entrypoint a little differently might do the job:

https://webpack.js.org/concepts/entry-points/

Reply:
Or this:
https://webpack.js.org/plugins/html-webpack-plugin/

Either way, I think the goal would be to achieve Mirage loading in fully before React starts, so that the React application code can remain as unchanged as possible. Ideally the application code does not have to make any accommodations for the possibility that Mirage is shimmed in.

Reply:
Ohh yess I actually tried to split the mirage JS chunk out but was worried about about execution order. But from that link above, seems like we can use `dependOn`.